test/integration/scheduler/preemption_test.go

```diff
@@ -366,6 +372,78 @@ func TestDisablePreemption(t *testing.T) {
 	}
 }
 
+func TestPodPriorityResolution(t *testing.T) {
+	admission := priority.NewPlugin()
+	context := initTestScheduler(t, initTestMaster(t, "preemption", admission), true, nil)
+	defer cleanupTest(t, context)
+	cs := context.clientSet
+
+	// Build clientset and informers for controllers.
+	externalClientset := kubernetes.NewForConfigOrDie(&restclient.Config{
+		QPS:           -1,
+		Host:          context.httpServer.URL,
+		ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}}})
+	externalInformers := informers.NewSharedInformerFactory(externalClientset, time.Second)
+	admission.SetExternalKubeClientSet(externalClientset)
+	admission.SetExternalKubeInformerFactory(externalInformers)
+	externalInformers.Start(context.stopCh)
+	externalInformers.WaitForCacheSync(context.stopCh)
+
+	tests := []struct {
+		PriorityClass    string
+		Pod              *v1.Pod
+		ExpectedPriority int32
+	}{
+		{
+			PriorityClass:    scheduling.SystemNodeCritical,
+			ExpectedPriority: scheduling.SystemCriticalPriority + 1000,
+			Pod: initPausePod(cs, &pausePodConfig{
+				Name:              fmt.Sprintf("pod1-%v", scheduling.SystemNodeCritical),
+				Namespace:         metav1.NamespaceSystem,
+				PriorityClassName: scheduling.SystemNodeCritical,
+			}),
+		},
+		{
+			PriorityClass:    scheduling.SystemClusterCritical,
+			ExpectedPriority: scheduling.SystemCriticalPriority,
+			Pod: initPausePod(cs, &pausePodConfig{
+				Name:              fmt.Sprintf("pod2-%v", scheduling.SystemClusterCritical),
+				Namespace:         metav1.NamespaceSystem,
+				PriorityClassName: scheduling.SystemClusterCritical,
+			}),
+		},
```

Comment:
Add a negative case? one for an unknown priority, and expect the scheduler to fail?

Reply:
Similar to my above reply, we're not setting integer values, we are setting a string and checking that it resolves to an integer, so I can't add a negative value check but I can add one for an unknown string and check that it resolves to nothing if you want. What do you think?

Reply:
Sorry, I didn't mean a negative number, just a negative case, i.e. a case that should be rejected.

Reply:
Ah, I see then I misunderstood. I added a case for an invalid priority that should be rejected